### ISSUE TITLE
Add magic filtering utility view

### ISSUE THREAD

**Issue by lewiscollard** [onespacemedia/project-template#80]
See: https://github.com/onespacemedia/owlstone-medical/blob/develop/owlstone/apps/utils/views.py

It basically allows you to declare the fields on which the model can be filtered in a ListView, and the view does everything automatically. It also sticks the filters into the context, so you just loop over `filters` in your context to list them (which allows you to have a single filtered-list-of-things template if you choose).

Basically every list view we've ever written has some kind of FK/M2M filtering and it saves a ton of writing nearly-identical ListView derivatives with very similar `get_queryset`, `get_context_data`,  etc.

**Comment by danielsamuels** [comment]
Sounds similar to [django-filter](https://github.com/carltongibson/django-filter), I wonder if there's any benefit of using one over the other.

**Comment by lewiscollard** [comment]
django-filter has the advantage of doing a lot more. But my totally unbiased opinion is that I like my implementation for the kind of simple FK/M2M filtering that comes up on most projects just because of its simplicity.
